framework/src/ble_manager/ble_manager_state.c

```diff
@@ -197,6 +267,137 @@ ble_result_e blemgr_handle_request(blemgr_msg_s *msg)
 		ret = ble_drv_conn_is_any_active(is_active);
 	} break;
 
+	// Scanner
+	case BLE_EVT_CMD_START_SCAN: {
+		BLE_STATE_CHECK;
+
+		if (g_scan_ctx.state != BLE_SCAN_STOPPED) {
+			ret = TRBLE_BUSY;
+			break;
+		}
+
+		ble_scan_state_e priv_state = g_scan_ctx.state;
+		blemgr_msg_params *param = (blemgr_msg_params *)msg->param;
+		ble_scan_filter *filter = (ble_scan_filter *)param->param[0];
+		ble_scan_callback_list *callbacks = (ble_scan_callback_list *)param->param[1];
+
+		if (filter != NULL) {
+			memcpy(&(g_scan_ctx.filter), filter, sizeof(ble_scan_filter));
+		}
+
+		if (callbacks == NULL) {
+			ret = TRBLE_INVALID_ARGS;
```

Comment:
In this case, switch-case will be finished.
Is it okay to copy the `filter` into the `g_scan_ctx.filter` even for this case?

Reply:
I fixed it

Reply:
How about moving into the line 278?
Because, for failure case, we don't need to set the local variables also.